### ISSUE TITLE
Another indentation problem with nested function calls

### ISSUE THREAD

**Issue by hgschmie** [pinterest/ktlint#1278]
## Expected Behavior

IntelliJ formats kotlin code (https://github.com/jdbi/jdbi/blob/master/kotlin/src/main/kotlin/org/jdbi/v3/core/kotlin/internal/KotlinPropertyArguments.kt)

like this:

```kotlin
    override fun getValue(name: String, ctx: StatementContext): Optional<TypedValue> {
        val property: KProperty1<*, *> = properties[name] ?: return Optional.empty()
        val mutableProperty = property as? KMutableProperty1
        val type = QualifiedType.of(property.returnType.javaType)
            .withAnnotations(
                getQualifiers(
                    kClass.primaryConstructor?.parameters?.find { it.name == name },
                    property,
                    property.getter,
                    mutableProperty?.setter,
                    mutableProperty?.setter?.parameters?.getOrNull(0)
                )
            )
        val value = property.getter.call(obj)
        return Optional.of(TypedValue(type, value))
    }
```

This looks pretty reasonable and readable to me. 

## Observed Behavior

Running `ktlint --experimental` yields
```
[ERROR] src/main/kotlin/org/jdbi/v3/core/kotlin/internal/KotlinPropertyArguments.kt:49:17: Unexpected indentation (expected 12, actual 16)
[ERROR] src/main/kotlin/org/jdbi/v3/core/kotlin/internal/KotlinPropertyArguments.kt:56:13: Unexpected indentation (expected 8, actual 12)
```

which refers to  `                getQualifiers(` and the closing brace below. Ktlint wants this on the same level as the `.withAnnotations` which makes the code less readable. IntelliJ agrees with me. :-) 

## Steps to Reproduce
put the code above in a kotlin file. Run `ktlint --experimental` on it.

## Your Environment
ktlint 0.42.1
Java version: 17.0.1, vendor: Homebrew, runtime: /usr/local/Cellar/openjdk/17.0.1/libexec/openjdk.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "11.6.1", arch: "x86_64", family: "mac"

**Comment by paul-dingemans** [comment]
Please try again with latest ktlint version 0.43.0 which contains fixes for some indentation problems.

**Comment by hgschmie** [comment]
I tried with the ktlint-maven-plugin (https://github.com/gantsign/ktlint-maven-plugin), which includes ktlint 0.43 and it shows the same problem.

**Comment by paul-dingemans** [comment]
That plugin currently runs on ktlint 0.42.1, not on 0.43.0

**Comment by hgschmie** [comment]
I tried with 1.11.0 (just released) which uses 0.43 and this fixes the problem.
